internal/providers/nscloud/registry.go

```diff
@@ -107,52 +97,42 @@ func (dk defaultKeychain) Resolve(ctx context.Context, r authn.Resource) (authn.
 		return authn.Anonymous, nil
 	}
 
-	ref, err := name.ParseReference(r.String())
-	if err != nil {
-		return nil, err
-	}
-
-	values := url.Values{}
-	values.Add("scope", fmt.Sprintf("repository:%s:push,pull", ref.Context().RepositoryStr()))
-	values.Add("service", "Authentication")
-
-	req, err := http.NewRequestWithContext(ctx, "GET", fmt.Sprintf("https://%s?%s", loginEndpoint, values.Encode()), nil)
-	if err != nil {
-		return nil, err
-	}
+	exchangeToken := func(ctx context.Context) (string, error) {
```

Comment:
Shall we define this in a shareable place so that we can also use it from https://github.com/namespacelabs/foundation/blob/7deff0f173d3a3c1ddc5606d649b8ebf2135e8c8/internal/providers/nscloud/api/rpc.go#L37-L38 ?

Reply:
Hm, I was thinking to move it to `auth` pkg, but then there will be an import cycle. What would be the better place? `interna;/fnapi`, `internal/providers/nscloud/api`?

Reply:
I'd start with `internal/providers/nscloud/api` since these tokens are nscloud specific.

Reply:
done